ground/src/main/java/com/google/android/ground/ui/map/gms/LocationOfInterestClusterManager.kt

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright 2022 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.android.ground.ui.map.gms
+
+import android.content.Context
+import com.google.android.gms.maps.GoogleMap
+import com.google.android.ground.model.geometry.Point
+import com.google.android.ground.model.locationofinterest.LocationOfInterest
+import com.google.android.ground.ui.map.MapLocationOfInterest
+import com.google.android.ground.util.toImmutableSet
+import com.google.maps.android.clustering.ClusterManager
+import timber.log.Timber
+
+class LocationOfInterestClusterManager(context: Context?, map: GoogleMap) :
```

Comment:
Can we abstract LOI somehow as mentioned earlier so that the map adapter needs to know less about specific business logic?

Reply:
We could in theory, but I'm a bit wary of introducing yet another abstraction. We can extend `MapLocationOfInterest` for this purpose and operate on those.

Reply:
The enforce stricter separation of concerns, the MapAdapter would deal exclusively with geometries and map-like objects rathe rthan Ground-specific things like "LOIs". Let's discuss offline.